lib/connection-manager.js

```diff
@@ -48,157 +56,98 @@ function restartSelf() {
             detached: true
         });
     }
-}
+};
 
-module.exports.connectLogServer = function(config, programName) {
-    var logurl = (config.LOGDNA_LOGSSL ? 'https://' : 'http://') + config.LOGDNA_LOGHOST + ':' + config.LOGDNA_LOGPORT;
-    var options = {
-        query: {
-            auth_token: config.auth_token
-            , timestamp: Date.now()
-            , compress: config.COMPRESS ? 1 : 0
-            , tailmode: config.TAIL_MODE !== 'trs' ? config.TAIL_MODE : undefined
-            , transport: config.TRANSPORT !== 'http' ? config.TRANSPORT : undefined
-        }
-        , reconnection: config.LOGDNA_RECONNECT
-        , perMessageDeflate: !!config.COMPRESS
+// Prepare Stats Object
+const prepareStats = (config) => {
+    let stats = {
+        e: 's'
+        , m: process.memoryUsage()
+        , b: config.bufferStats
     };
-    var proxy = getProxyFromURI(url.parse(logurl));
-    if (proxy) options.agent = new HttpsProxyAgent(url.parse(proxy));
 
-    var firstrun = true;
-    debug(logurl);
-    var socket = new WebSocket(logurl, options);
-    linebuffer.setSocket(socket);
-    debug(socket.options);
-    return new Promise(resolve => {
-        debug('connecting to websocket');
-        socket.on('open', function() {
-            log('Connected to ' + config.LOGDNA_LOGHOST + ':' + config.LOGDNA_LOGPORT + ' (' + socket._socket.remoteAddress + ')' + (config.LOGDNA_LOGSSL ? ' (SSL)' : '') + (proxy ? ' via ' + proxy : ''));
-
-            if (firstrun) {
-                firstrun = false;
-
-                // start streaming logs from logdir(s) on startup
-                fileUtils.streamAllLogs(config);
-
-                if (os.platform() === 'win32' && config.winevent) {
-                    if (typeof config.winevent === 'string') {
-                        config.winevent = config.winevent.split(',');
-                    }
-
-                    winUtils.streamEventLog({
-                        events: config.winevent
-                    }, socket);
-
-                }
+    if (stats.b && stats.b.ts) {
+        stats.b.ms = Date.now() - stats.b.ts;
+    }
 
-                setTimeout(exports.sendStats, config.STATS_INTERVAL, config, socket);
-                setTimeout(exports.socketKeepAlive, config.SOCKET_KEEPALIVE, config, socket);
+    return stats;
+};
 
-            } else {
-                // reconnected, resume streaming
-                debug(fileUtils.files);
-                log('Streaming resumed: ' + fileUtils.files.length + ' file(s)');
+// Send POST Request to HealthCheck Endpoint
+const checkHealth = (config) => {
+    return async.retry({
+        times: 3
+        , interval: (retryCount) => {
+            return 50 * Math.pow(2, retryCount);
+        }
+        , errorFilter: (err) => {
+            return config.DEFAULT_HTTP_ERRORS && config.DEFAULT_HTTP_ERRORS.includes(err.code);
+        }
+    }, (reqCallback) => {
+        return req.post(logurl + config.LOGDNA_HEALTHCHECK, {
```

Comment:
One of the reasons is I don't wanna repeat writing/using the same set of variables instead of one variable but `config.LOGDNA_LOGURL` is gonna solve this too.

Reply:
that's true, that works too